### ISSUE TITLE
45005773: Xcode 10 repeatedly crashing due to UI access on main thread

### ISSUE THREAD

**Issue by openradar-mirror** [lionheart/openradar-mirror#20517]
#### Description

Summary:
Xcode 10 regularly crashes during use with the following exception: CoreAnimation: CA_ASSERT_MAIN_THREAD_TRANSACTIONS is set and an implicit transaction wasn't created on a main thread. The crash occurs on Dispatch queue: com.apple.NSXPCConnection.user.com.apple.dt.Xcode.sourcecontrol.Git. 

Steps to Reproduce:
Unknown. Might be related to the Xcode commit UI. 

Expected Results:
Xcode should not access AppKit on a background thread. 

Actual Results:
Xcode accesses AppKit on a background thread. 

Version:
10

Notes:
See attached crash report. Might only occur on touch-bar macs. Did also not occur before upgrading to macOS Mojave.

-
Product Version: 10
Created: 2018-10-04T12:24:49.155880
Originated: 2018-10-04T14:24:00
Open Radar Link: http://www.openradar.me/45005773

**Comment by picnicbob** [comment]
Seems to be due to CATiledLayer in my case

**Comment by tanis2000** [comment]
@picnicbob it's happening to me as well for the same reason. Did you find a solution?
I'm also trying to find out who is calling my view's drawRect from the CA DispatchGroup thread as I'm seeing the same behaviour from within the app.. :-P

**Comment by picnicbob** [comment]
@tanis2000 No, I have not found a solution. CATiledLayer is designed to call -drawRect: on a background thread, so this is expected CoreAnimation behavior. My guess is that Xcode itself has CA_ASSERT_MAIN_THREAD_TRANSACTIONS enabled when it attempts to render the view hierarchy which causes it to crash.

**Comment by tanis2000** [comment]
That's not smart. I wonder if Apple will ever take a look at it. It's painful not being able to debug views this way.

**Comment by 0xcaff** [comment]
This seems to also happen if you use material tiling through `SCNMaterialProperty`'s `wrapS`/`wrapT` also.
